### PR TITLE
Refactor mitochondrial analysis workflow

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

This PR,

- Breaks down mitochondrial analysis workflow into smaller subworkflows that are more modular. 
- replaces the parameter `skip_mt_analysis` which was used to turn on/off the mitochondrial workflow
- adds a new parameter `skip_mt_annotation` which can be used to turn on/off annotation and ranking for mitochondrial SNVs

With changes in this PR, mitochondrial SNVs will no longer be called by deepvariant, and they will come from the GATKs best practices workflow for mitochondria.

#

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test_one_sample,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ nf-core/raredisease consists of several tools used for various purposes. For con
 5. Variant calling - Structural variants (SV) (Tiddit & Manta)
 6. SNV annotation & ranking (rohcall, vcfanno, ensembl VEP, GENMOD)
 7. SV annotation & ranking (SVDB query, ensembl VEP, GENMOD)
-8. Mitochondrial analysis
+8. Mitochondrial annotation
```
Comment (Lucpen) [inline docs/usage.md:137]: Mitochondrial annotation is 8 here, but was 9 before (line 23) and is 9 later on (line 245), check if it's correct

**Review by Lucpen — COMMENTED** [review]
Found one tiny thing, I will take another look before approving

File: subworkflows/local/call_snv.nf [hunk 2]
```diff
@@ -2,53 +2,107 @@
...
 
     emit:
-        vcf      = ch_vcf      // channel: [ val(meta), path(vcf) ]
-        tabix    = ch_tabix    // channel: [ val(meta), path(tbi) ]
-        versions = ch_versions // channel: [ path(versions.yml) ]
+        genome_vcf       = ch_genome_vcf                // channel: [ val(meta), path(vcf) ]
+        genome_tabix     = ch_genome_tabix              // channel: [ val(meta), path(tbi) ]
+        genome_vcf_tabix = ch_genome_vcf_tabix          // channel: [ val(meta), path(vcf), path(tbi) ]
+        mt_vcf           = POSTPROCESS_MT_CALLS.out.vcf // channel: [ val(meta), path(vcf) ]
+        mt_tabix         = POSTPROCESS_MT_CALLS.out.tbi // channel: [ val(meta), path(vcf) ]
```
Comment (jemten) [inline subworkflows/local/call_snv.nf:106]:
```suggestion
        mt_tabix         = POSTPROCESS_MT_CALLS.out.tbi // channel: [ val(meta), path(tbi) ]
```

File: subworkflows/local/call_structural_variants.nf [hunk 3]
```diff
@@ -28,21 +33,25 @@ workflow CALL_STRUCTURAL_VARIANTS {
...
 
-        CALL_SV_GERMLINECNVCALLER (ch_bam_bai, ch_genome_fasta, ch_genome_fai, ch_readcount_intervals, ch_genome_dictionary, ch_ploidy_model, ch_gcnvcaller_model)
+        CALL_SV_GERMLINECNVCALLER (ch_genome_bam_bai, ch_genome_fasta, ch_genome_fai, ch_readcount_intervals, ch_genome_dictionary, ch_ploidy_model, ch_gcnvcaller_model)
             .genotyped_intervals_vcf
             .collect{it[1]}
             .set { gcnvcaller_vcf }
 
+        CALL_SV_MT (ch_mt_bam_bai, ch_genome_fasta)
+
+//        CALL_SV_MT_SHIFT (ch_mtshift_bam_bai, ch_mtshift_fasta)
```
Comment (jemten) [inline subworkflows/local/call_structural_variants.nf:53]: remove?

File: subworkflows/local/variant_calling/postprocess_MT_calls.nf [hunk 4]
```diff
@@ -0,0 +1,111 @@
+//
+// Merge and annotate MT
```
Comment (jemten) [inline subworkflows/local/variant_calling/postprocess_MT_calls.nf:2]:
```suggestion
// Merge MT variants
```

File: subworkflows/local/variant_calling/postprocess_MT_calls.nf [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        ch_genome_fasta        // channel: [mandatory] [ val(meta), path(fasta) ]
+        ch_genome_dictionary   // channel: [mandatory] [ val(meta), path(dict) ]
+        ch_genome_fai          // channel: [mandatory] [ val(meta), path(fai) ]
+        ch_mtshift_backchain   // channel: [mandatory] [ val(meta), path(backchain) ]
+        ch_case_info           // channel: [mandatory] [ val(case_info) ]
+
+    main:
+        ch_versions = Channel.empty()
+
+        // LIFTOVER VCF FROM REFERENCE MT TO SHIFTED MT
```
Comment (jemten) [inline subworkflows/local/variant_calling/postprocess_MT_calls.nf:28]:
```suggestion
        // LIFTOVER SHIFTED VCF TO REFERENCE MT POSITIONS
```

File: subworkflows/local/annotate_mt_snvs.nf [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        ch_mt_vcf
+            .combine(ANNOTATE_CADD.out.vcf.ifEmpty("null"))
+            .branch { it  ->
+                merged: it[2].equals("null")
+                    return [it[0], it[1]]
+                cadd: !(it[2].equals("null"))
+                    return [it[2], it[3]]
+            }
+            .set { ch_for_mix }
+        ch_vep_in = ch_for_mix.merged.mix(ch_for_mix.cadd)
```
Comment (jemten) [inline subworkflows/local/annotate_mt_snvs.nf:47]: Is there reason to use the same type of channel combinations as in https://github.com/nf-core/raredisease/pull/421 ?

**Review by jemten — COMMENTED** [review]
Massive work Ram. Some small comments and questions

**Review by jemten — APPROVED** [review]
Looks good

**Pull request merged**